odoo/models.py

```diff
@@ -4870,6 +4870,14 @@ def with_prefetch(self, prefetch_ids=None):
             prefetch_ids = self._ids
         return self._browse(self.env, self._ids, prefetch_ids)
 
+    def prefetch_fields(self, fields):
+        """Prefetch fields manually in case they are missing.
+
+        :param list fields: List of fields to be prefetched in current recordset.
+        """
+        _fields = [field for field in fields if field not in self._cache]
```

Comment:
```python
if fields:
     self.browse(self.ids).read(fields - self._cache.keys())
```

 no ?

Reply:
```python
>>> ["field_1", "field_2"] - ["field_1"]                                                                                                                                                                             
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'list' and 'list'

unsupported operand type(s) for -: 'list' and 'list'
```

Reply:
I added the if, though.

Reply:
`fields = list(set(fields) - set(self._cache.keys())`

Reply:
Current algorithm is twice as fast, and IMHO easier to read:

```python
>>> import timeit                                                                                                                                                                 

>>> fields = ["field1", "field2", "field3", "field4", "field5"]                                                                                                                   

>>> cache = {"field1": True, "field6": True}                                           
                                                                                           
>>> timeit.timeit("[f for f in fields if f not in cache]", number=1000000, globals=locals())                                                                                      
1.2588749449996612

>>> timeit.timeit("list(set(fields) - set(cache.keys()))", number=1000000, globals=locals())                                                                                      
2.607388816999901
```

Reply:
Tanks, I will know now